packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -1068,8 +1077,35 @@ Future<void> refreshPluginsList(
   }
 }
 
+/// Injects plugins found in `pubspec.yaml` into the platform-specific projects
+/// only at build-time.
+///
+/// This method is similar to [injectPlugins], but used only for platforms where
+/// the plugin files are not required at compile-time (currently: Web).
+///
+/// It requires a `destination` parameter to specify the root directory (in the
+/// filesystem or memory-fs) of current build output.
+Future<void> injectBuildTimePluginFiles(
+  FlutterProject project, {
+  required Directory destination,
+  bool webPlatform = false,
+}) async {
+  final List<Plugin> plugins = await findPlugins(project);
+  // Sort the plugins by name to keep ordering stable in generated files.
+  plugins.sort((Plugin left, Plugin right) => left.name.compareTo(right.name));
+  if (webPlatform) {
+    await _writeWebPluginRegistrant(project, plugins, destination);
+  }
+}
+
 /// Injects plugins found in `pubspec.yaml` into the platform-specific projects.
 ///
+/// These files are required at compile-time.
+///
+/// This method used to generate the plugin registrar for the webPlatform too,
```

Comment:
Rather than say what used to happen (which is probably not interesting to most readers), I would just focus on the explanation of why web isn't here.

Reply:
I rewrote these two bits, I think the difference is more clear now (or somebody who's actually reading these docs will have a better context of what's going on)